core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -746,6 +752,24 @@ private[spark] class AppStatusListener(
         maybeUpdate(esummary, now)
       }
 
+      val speculationStageSummary = stage.speculationStageSummary
+      if (event.taskInfo.speculative) {
+        speculationStageSummary.numActiveTasks -= 1
+        speculationStageSummary.numCompletedTasks += completedDelta
+        speculationStageSummary.numFailedTasks += failedDelta
+        speculationStageSummary.numKilledTasks += killedDelta
+      }
+      speculationStageSummary.killedTasksSummary = killedTaskSummaryForSpeculationStageSummary(
+        event.reason,
+        speculationStageSummary.killedTasksSummary,
+        event.taskInfo.speculative)
+
+      if (isLastTask) {
```

Comment:
`speculationsStageSummary` is optional so we should update it only for speculative tasks.